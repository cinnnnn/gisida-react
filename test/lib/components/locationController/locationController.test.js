
import React from 'react';
import LocationController from "../../../../src/lib/components/LocationController/LocationController";
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

const initialState = {
    LOC: {
        active: true,
        location: {},
        locations: {
            locA: {label: "test 1"},
            locB: {label: "test 2"}
        }
    }
 };

const mockStore = configureStore()
const store = mockStore(initialState);
const dispatch = jest.spyOn(store, 'dispatch');

const componentWrapper = mount(
	<LocationController 
        store={store}
        mapId="map-1"
        dispatch={dispatch}
	/>
);

describe('LocationController', () => {

    it('FilterSelector component renders correctly', () => {
		const json = toJson(componentWrapper);
		expect(json).toMatchSnapshot();
    })
    
    it('should call onLocationClick on click and dispatch', () => {
        componentWrapper.find('a').at(0).simulate('click');
        expect(dispatch).toHaveBeenCalledTimes(1);
    })

})