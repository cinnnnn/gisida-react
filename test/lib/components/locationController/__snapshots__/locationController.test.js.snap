// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationController FilterSelector component renders correctly 1`] = `
<Connect(LocationController)
  dispatch={[MockFunction]}
  mapId="map-1"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LocationController
    dispatch={[MockFunction]}
    mapId="map-1"
    state={
      Object {
        "active": true,
        "location": Object {},
        "locations": Object {
          "locA": Object {
            "label": "test 1",
          },
          "locB": Object {
            "label": "test 2",
          },
        },
        "mapId": "map-1",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <section
      className="location-controller"
    >
      <h4
        className="loc-title"
      >
        Map Locations
      </h4>
      <ul
        className="loc-list"
      >
        <li
          className="loc-item"
          key="0"
        >
          <a
            alt="Center on test 1"
            className="loc-link"
            data-id="locA"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "normal",
              }
            }
            title="Center on test 1"
          >
            test 1
          </a>
        </li>
        <li
          className="loc-item"
          key="1"
        >
          <a
            alt="Center on test 2"
            className="loc-link"
            data-id="locB"
            onClick={[Function]}
            style={
              Object {
                "fontWeight": "normal",
              }
            }
            title="Center on test 2"
          >
            test 2
          </a>
        </li>
      </ul>
    </section>
  </LocationController>
</Connect(LocationController)>
`;
